tests/e2e/tikv/tikv.go

```diff
@@ -0,0 +1,546 @@
+		finalDeleteSlots:   sets.NewInt32(2),
+		scaleInOrdinals:    []int32{5, 4, 2},
+		deleteSlotsEvolutions: [][]int32{
+			{1, 6},
+			{2},
+		},
+		scaleInGroups: [][]int32{
+			{5, 4},
+			{2},
+		},
+	},
+	{
+		idx:                9,
+		originTiKVReplica:  5,
+		finalTiKVReplica:   3,
+		scaleInParallelism: 2,
+		originDeleteSlots:  sets.NewInt32(1, 6),
+		finalDeleteSlots:   sets.NewInt32(2, 7),
+		scaleInOrdinals:    []int32{5, 4, 2},
+		deleteSlotsEvolutions: [][]int32{
+			{1, 6},
+			{7},
+			{2, 7},
+		},
+		scaleInGroups: [][]int32{
+			{5, 4},
+			{2},
+		},
+	},
+}
+
+func (t *testcase) description() string {
+	return fmt.Sprintf("[origin-%v-final-%v-parallelism-%v-testcase-idx-%v]", t.originTiKVReplica, t.finalTiKVReplica, t.scaleInParallelism, t.idx)
+}
+
+var _ = ginkgo.Describe("[TiKV: Scale in simultaneously]", func() {
+	f := e2eframework.NewDefaultFramework("tidb-cluster-with-asts")
+
+	var ns string
+	var c clientset.Interface
+	var cli versioned.Interface
+	var asCli asclientset.Interface
+	var aggrCli aggregatorclient.Interface
+	var apiExtCli apiextensionsclientset.Interface
+	var cfg *tests.Config
+	var config *restclient.Config
+	var fwCancel context.CancelFunc
+	var fw portforward.PortForward
+
+	ginkgo.BeforeEach(func() {
+		ns = f.Namespace.Name
+		c = f.ClientSet
+
+		var err error
+		config, err = framework.LoadConfig()
+		framework.ExpectNoError(err, "failed to load config")
+		cli, err = versioned.NewForConfig(config)
+		framework.ExpectNoError(err, "failed to create clientset for pingcap")
+		asCli, err = asclientset.NewForConfig(config)
+		framework.ExpectNoError(err, "failed to create clientset for advanced-statefulset")
+		aggrCli, err = aggregatorclient.NewForConfig(config)
+		framework.ExpectNoError(err, "failed to create clientset kube-aggregator")
+		apiExtCli, err = apiextensionsclientset.NewForConfig(config)
+		framework.ExpectNoError(err, "failed to create clientset apiextensions-apiserver")
+		clientRawConfig, err := e2econfig.LoadClientRawConfig()
+		framework.ExpectNoError(err, "failed to load raw config for tidb-operator")
+		ctx, cancel := context.WithCancel(context.Background())
+		fw, err = portforward.NewPortForwarder(ctx, e2econfig.NewSimpleRESTClientGetter(clientRawConfig))
+		framework.ExpectNoError(err, "failed to create port forwarder")
+		fwCancel = cancel
+		cfg = e2econfig.TestConfig
+	})
+
+	ginkgo.AfterEach(func() {
+		if fwCancel != nil {
+			fwCancel()
+		}
+	})
+
+	ginkgo.Context("[Scale in simultaneously without asts]", func() {
+		var ocfg *tests.OperatorConfig
+		var oa *tests.OperatorActions
+		var genericCli ctrlCli.Client
+		var err error
+
+		ginkgo.BeforeEach(func() {
+			ocfg = e2econfig.NewDefaultOperatorConfig(cfg)
+			oa = tests.NewOperatorActions(cli, c, asCli, aggrCli, apiExtCli, tests.DefaultPollInterval, ocfg, e2econfig.TestConfig, fw, f)
+			oa.CleanCRDOrDie()
+			oa.CreateCRDOrDie(ocfg)
+			ginkgo.By("Installing tidb-operator")
+			oa.CleanOperatorOrDie(ocfg)
```

Comment:
Do we need to recreate TiDB Operator for each case? cc @KanShiori

Reply:
no, because there is a new cluster deployed for each case.

Reply:
OK